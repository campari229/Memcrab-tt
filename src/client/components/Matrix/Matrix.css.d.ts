declare const styles: {
  readonly "table": string;
  readonly "cell": string;
  readonly "body": string;
  readonly "row": string;
  readonly "cellClosest": string;
  readonly "cellAvarange": string;
  readonly "rowFooter": string;
  readonly "text": string;
  readonly "buttonWrapper": string;
  readonly "buttonRemove": string;
  readonly "button-new": string;
};
export = styles;

