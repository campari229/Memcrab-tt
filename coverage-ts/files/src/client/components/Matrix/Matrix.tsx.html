
  <!DOCTYPE html>
  <html>
    <head>
      <title>Matrix.tsx</title>
      <link href="https://cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" type="text/css" rel="stylesheet">
      <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.js" type="text/javascript" charset="utf-8"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/mode/javascript/javascript.min.js" type="text/javascript" charset="utf-8"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.css" type="text/css" rel="stylesheet">
<script src="..\..\..\..\..\assets\source-file.js" type="text/javascript" charset="utf-8"></script>
<link href="..\..\..\..\..\assets\source-file.css" type="text/css" rel="stylesheet">
    </head>
    <body>
    <div style="margin-top:3em" class="ui container"><h1 class="ui header"><a href="..\..\..\..\..\index.html">TypeScript coverage report</a></h1><table style="margin-top:2em" class="ui celled table"><thead class=""><tr class=""><th class="">Filename</th><th class="">Percent</th><th class="">Threshold</th><th class="">Total</th><th class="">Covered</th><th class="">Uncovered</th></tr></thead><tbody class=""><tr class="positive"><td class="">src\client\components\Matrix\Matrix.tsx</td><td class="">100.00%</td><td class="">80%</td><td class="">160</td><td class="">160</td><td class="">0</td></tr></tbody></table><textarea id="editor" readonly="" style="margin-top:3em">import React from &#x27;react&#x27;;
import { useDispatch, useSelector } from &#x27;react-redux&#x27;;
import {
  getCells,
  getTableFooter,
  addRow,
  removeRow,
  increment,
  percentsToggle,
  showClosest
} from &#x27;../../store/store&#x27;;
import { rowCreator } from &#x27;../../lib/lib&#x27;;
import { Cell } from &#x27;../../../Interfaces&#x27;

import &#x27;./Matrix.scss&#x27;;

export const Matrix = () =&gt; {
  const dispatch = useDispatch();
  const cells = useSelector(getCells);
  const tableFooter = useSelector(getTableFooter);

  if (cells) {
    return (
      &lt;table className=&quot;table&quot;&gt;
        &lt;tbody className=&quot;table__body&quot;&gt;
          {cells.map((row: Cell[], i: number)=&gt; (
            &lt;tr className=&quot;table__row&quot; key={i}&gt;
              &lt;td className=&quot;table__button-wrapper&quot;&gt;
                &lt;button className=&quot;table__button-remove&quot; onClick={() =&gt; dispatch(removeRow(i))}&gt;Remove row&lt;/button&gt;
              &lt;/td&gt;
              {row.map(cell =&gt; (
                &lt;td
                  key={cell.id}
                  onClick={() =&gt; dispatch(increment(cell.id, i))}
                  onMouseOver={() =&gt; dispatch(showClosest(cell))}
                  onMouseOut={() =&gt; dispatch(showClosest(cell))}
                  className={cell.isCloser ? &#x27;table__cell table__cell-closest&#x27; : &#x27;table__cell&#x27;}
                &gt;
                  &lt;p className=&quot;table__text&quot;&gt;
                    {cell.isPercentsShown
                      ? `${Math.round((cell.amount / row.reduce((acumulator, cell) =&gt; acumulator + cell.amount, 0)) * 100)}%`
                      : cell.amount
                    }
                  &lt;/p&gt;
                  &lt;div
                    className=&quot;table__percents&quot;
                    style={
                      cell.isPercentsShown
                        ? {height: `${Math.round((cell.amount / row.reduce((acumulator, cell) =&gt; acumulator + cell.amount, 0)) * 100)}%`, backgroundColor: &quot;red&quot;}
                        : {height: &quot;0&quot;}
                    }
                  /&gt;
                &lt;/td&gt;
              ))}
              &lt;td
                className=&quot;table__cell&quot;
                onMouseOver={() =&gt; dispatch(percentsToggle(i))}
                onMouseOut={() =&gt; dispatch(percentsToggle(i))}
              &gt;
                {row.reduce((acumulator, cell) =&gt; acumulator + cell.amount, 0)}
              &lt;/td&gt;
            &lt;/tr&gt;
          ))}
        &lt;/tbody&gt;
        &lt;tfoot className=&quot;table__footer&quot;&gt;
          &lt;tr className =&quot;table__row-footer&quot;&gt;
            {tableFooter.map((cell: Cell) =&gt; (
              &lt;td key={cell.id} className=&quot;table__cell-avarange&quot;&gt;
                {cell.amount}
              &lt;/td&gt;
            ))}
            &lt;td className=&quot;table__button-wrapper&quot;&gt;
              &lt;button className=&quot;table__button-new&quot; onClick={() =&gt; dispatch(addRow(rowCreator(cells[0].length)))}&gt;add row&lt;/button&gt;
            &lt;/td&gt;
          &lt;/tr&gt;
        &lt;/tfoot&gt;
      &lt;/table&gt;
    )
  } else {
    return &lt;h2&gt;oops&lt;/h2&gt;
  }
}

</textarea><pre id="annotations" style="display:none">[]</pre></div>
    <p class="footer-text">TypeScript Coverage Report generated by <a href="https://github.com/plantain-00/type-coverage">type-coverage</a> and <a href="https://github.com/alexcanessa/typescript-coverage-report">typescript-coverage-report</a> at Wed, 04 Nov 2020 16:03:48 GMT</p>
    </body>
  </html>
  