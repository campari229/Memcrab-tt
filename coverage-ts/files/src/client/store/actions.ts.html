
  <!DOCTYPE html>
  <html>
    <head>
      <title>actions.ts</title>
      <link href="https://cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" type="text/css" rel="stylesheet">
      <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.js" type="text/javascript" charset="utf-8"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/mode/javascript/javascript.min.js" type="text/javascript" charset="utf-8"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.52.2/codemirror.min.css" type="text/css" rel="stylesheet">
<script src="..\..\..\..\assets\source-file.js" type="text/javascript" charset="utf-8"></script>
<link href="..\..\..\..\assets\source-file.css" type="text/css" rel="stylesheet">
    </head>
    <body>
    <div style="margin-top:3em" class="ui container"><h1 class="ui header"><a href="..\..\..\..\index.html">TypeScript coverage report</a></h1><table style="margin-top:2em" class="ui celled table"><thead class=""><tr class=""><th class="">Filename</th><th class="">Percent</th><th class="">Threshold</th><th class="">Total</th><th class="">Covered</th><th class="">Uncovered</th></tr></thead><tbody class=""><tr class="positive"><td class="">src\client\store\actions.ts</td><td class="">100.00%</td><td class="">80%</td><td class="">57</td><td class="">57</td><td class="">0</td></tr></tbody></table><textarea id="editor" readonly="" style="margin-top:3em">import { Cell } from &#x27;../../Interfaces&#x27;;

export const actions = {
  SET_CELLS: &#x27;SET_CELLS&#x27;,
  ADD_ROW: &#x27;ADD_ROW&#x27;,
  REMOVE_ROW: &#x27;REMOVE_ROW&#x27;,
  INCREMENT: &#x27;INCREMENT&#x27;,
  PERCENTS_TOGGLE: &#x27;PERCENTS_TOGGLE&#x27;,
  SET_NUMBER_OF_CLOSEST: &#x27;SET_NUMBER_OF_CLOSEST&#x27;,
  SHOW_CLOSEST: &#x27;SHOW_CLOSEST&#x27;
} as const

interface SetCells {
  type: typeof actions.SET_CELLS;
  cells: Cell[][];
}

interface AddRow {
  type: typeof actions.ADD_ROW;
  row: Cell[]
}

interface RemoveRow {
  type: typeof actions.REMOVE_ROW;
  number: number;
}

interface Increment {
  type: typeof actions.INCREMENT;
  id: number;
  rowIndex: number;
}

interface PercentsToggle {
  type: typeof actions.PERCENTS_TOGGLE;
  rowIndex: number;
}

interface ShowClosest {
  type: typeof actions.SHOW_CLOSEST;
  target: Cell;
}

interface SetNumberOfClosest {
  type: typeof actions.SET_NUMBER_OF_CLOSEST;
  numberOfClosest: number;
}

export type ActionType = SetCells | AddRow | RemoveRow | Increment | PercentsToggle | ShowClosest | SetNumberOfClosest</textarea><pre id="annotations" style="display:none">[]</pre></div>
    <p class="footer-text">TypeScript Coverage Report generated by <a href="https://github.com/plantain-00/type-coverage">type-coverage</a> and <a href="https://github.com/alexcanessa/typescript-coverage-report">typescript-coverage-report</a> at Wed, 04 Nov 2020 16:03:47 GMT</p>
    </body>
  </html>
  