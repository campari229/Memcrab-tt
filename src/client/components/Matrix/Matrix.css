.table {
  margin: 100px auto;
  width: 95vw;
}

.cell {
  position: relative;
  box-sizing: border-box;
  transition: transform 0.5s, background-color 0.5s;
  cursor: pointer;
  display: block;
  line-height: 80px;

  margin: 10px 10px 0 0;
  background-color: black;
  color: #fff;
  text-align: center;
  width: 80px;
  height: 80px;
  box-sizing: border-box;
};

.body {
  margin: 0 auto;
  display: block;
}

.row {
  display: flex;
}

.cell:active {
  transform: scale(0.9);
}

.cellClosest {
  position: relative;
  box-sizing: border-box;
  transition: transform 0.5s, background-color 0.5s;
  cursor: pointer;
  display: block;
  line-height: 80px;

  margin: 10px 10px 0 0;
  background-color: black;
  color: #fff;
  text-align: center;
  width: 80px;
  height: 80px;
  box-sizing: border-box;

  background-color: green;
  transform: scale(1.1);
}

.cellAvarange {
  margin: 10px 10px 0 0;
  background-color: black;
  color: #fff;
  text-align: center;
  width: 80px;
  height: 80px;
  box-sizing: border-box;
  display: block;
  line-height: 80px;
}

.rowFooter {
  transform: translateX(70px);
  display: flex;
}

.text {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;

  line-height: 80px;
  z-index: 2;
  margin: 0;
};

/* .x {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  z-index: 1;
  
  background-color: red;
  transition: height 0.5s;
} */

.buttonWrapper {
  background-color: #fff;
  display: flex;
}

.buttonRemove {
  background-color: black;
  color: #fff;

  width: 70px;
  height: 50px;
  margin: auto;

  border-radius: 0 50% 50% 0;
  cursor: pointer;
}

.button-new {
  width: 60px;
  height: 60px;

  background-color: black;
  color: #fff;

  border-radius: 50% 50% 0 0;
  cursor: pointer;
}
