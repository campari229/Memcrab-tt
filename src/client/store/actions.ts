import { Cell } from '../../Interfaces';

export const actions = {
  SET_CELLS: 'SET_CELLS',
  ADD_ROW: 'ADD_ROW',
  REMOVE_ROW: 'REMOVE_ROW',
  INCREMENT: 'INCREMENT',
  PERCENTS_TOGGLE: 'PERCENTS_TOGGLE',
  SET_NUMBER_OF_CLOSEST: 'SET_NUMBER_OF_CLOSEST',
  SHOW_CLOSEST: 'SHOW_CLOSEST'
} as const

interface SetCells {
  type: typeof actions.SET_CELLS;
  cells: Cell[][];
}

interface AddRow {
  type: typeof actions.ADD_ROW;
  row: Cell[]
}

interface RemoveRow {
  type: typeof actions.REMOVE_ROW;
  number: number;
}

interface Increment {
  type: typeof actions.INCREMENT;
  id: number;
  rowIndex: number;
}

interface PercentsToggle {
  type: typeof actions.PERCENTS_TOGGLE;
  rowIndex: number;
}

interface ShowClosest {
  type: typeof actions.SHOW_CLOSEST;
  target: Cell;
}

interface SetNumberOfClosest {
  type: typeof actions.SET_NUMBER_OF_CLOSEST;
  numberOfClosest: number;
}

export type ActionType = SetCells | AddRow | RemoveRow | Increment | PercentsToggle | ShowClosest | SetNumberOfClosest