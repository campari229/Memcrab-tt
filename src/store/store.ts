import {createStore, AnyAction } from 'redux';
import { Cell } from '../Interfaces';
import { getAverageValues } from '../lib/CellsCreator';

const SET_CELLS = 'SET_CELLS';
const ADD_ROW = 'ADD_ROW';
const REMOVE_ROW = 'REMOVE_ROW';

export const setCells = (cells: Cell[][]) => ({
  type: SET_CELLS,
  cells,
});
export const addRow = (row: Cell[]) => ({
  type: ADD_ROW,
  row,
})
export const removeRow = (number: number) => ({
  type: REMOVE_ROW,
  number,
})

export const getCells = (store: InitialState) => store.cells;
export const getTableFooter = (store: InitialState) => store.tableFooter;


type InitialState = {
  cells: Cell[][];
  tableFooter: Cell[];
};

const initialState = {
  cells: [],
  tableFooter: [],
};



const reducer = (store: InitialState = initialState, action: AnyAction ) => {
  switch (action.type) {
    case 'SET_CELLS':
      console.log(action.cells)
      console.log(getAverageValues(action.cells))
      return {
        ...store,
        cells: action.cells,
        tableFooter: getAverageValues(action.cells),
      };

    case 'ADD_ROW':
      console.log([...store.cells, action.row])
      return {
        ...store,
        cells: [
          ...store.cells,
          action.row,
        ],
        tableFooter: getAverageValues([...store.cells, action.row])
      };

    case 'REMOVE_ROW':
      const filteredCells = store.cells.filter((row, index) => index !== action.number)
      return {
        ...store,
        cells: filteredCells,
        tableFooter: getAverageValues(filteredCells),
      }

    default:
      return store;
  }
};

const store = createStore(reducer);

export default store;
