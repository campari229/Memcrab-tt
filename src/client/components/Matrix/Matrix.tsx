import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  getCells,
  getTableFooter,
  addRow,
  removeRow,
  increment,
  percentsToggle,
  showClosest
} from '../../store/store';
import { rowCreator } from '../../lib/lib';

import './Matrix.scss';
import { Row } from '../Row/Row';

export const Matrix = () => {
  const dispatch = useDispatch();
  const cells = useSelector(getCells);
  const tableFooter = useSelector(getTableFooter);

  if (cells) {
    return (
      <table className="table">
        <tbody className="table__body">
          {cells.map((row, i)=> (
            <tr className="table__row" key={i}>
              <Row row={row} i={i} />
            </tr>
          ))}
        </tbody>
        <tfoot className="table__footer">
          <tr className ="table__row-footer">
            {tableFooter.map((cell) => (
              <td key={cell.id} className="table__cell-avarange">
                {cell.amount}
              </td>
            ))}
            <td className="table__button-wrapper">
              <button className="table__button-new" onClick={() => dispatch(addRow(rowCreator(cells[0].length)))}>add row</button>
            </td>
          </tr>
        </tfoot>
      </table>
    )
  } else {
    return <h2>oops</h2>
  }
}

