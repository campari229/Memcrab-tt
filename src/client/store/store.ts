import React from 'react';
import { getAverageValues, findClosest } from '../lib/lib';
import { CellInterface } from '../../Interfaces';
import { actions, ActionType } from './actions';

export const setCells = (cells: CellInterface[][]) => ({
  type: actions.SET_CELLS,
  cells,
});
export const addRow = (row: CellInterface[]) => ({
  type: actions.ADD_ROW,
  row,
})
export const removeRow = (number: number) => ({
  type: actions.REMOVE_ROW,
  number,
})
export const increment = (id: number, rowIndex: number) => ({
  type: actions.INCREMENT,
  id,
  rowIndex,
})
export const percentsToggle = (rowIndex: number) => ({
  type: actions.PERCENTS_TOGGLE,
  rowIndex,
})
export const showClosest = (targetId: number) => ({
  type: actions.SHOW_CLOSEST,
  targetId,
})
export const setNumberOfClosest = (numberOfClosest: number) => ({
  type: actions.SET_NUMBER_OF_CLOSEST,
  numberOfClosest,
})

export type State = {
  cells: CellInterface[][];
  tableFooter: CellInterface[];
  numberOfClosest: number;
};

export const getCells = (state: State) => state.cells;
export const getTableFooter = (state: State) => state.tableFooter;

const initialState: State = {
  cells: [],
  tableFooter: [],
  numberOfClosest: 0,
};

export const reducer = (state: State = initialState, action: ActionType) => {
  switch (action.type) {
    case 'SET_CELLS':
      return {
        ...state,
        cells: action.cells,
        tableFooter: getAverageValues(action.cells),
      };

    case 'ADD_ROW':
      return {
        ...state,
        cells: [
          ...state.cells,
          action.row,
        ],
        tableFooter: getAverageValues([...state.cells, action.row])
      };

    case 'REMOVE_ROW':
      const filteredCells = state.cells.filter((row, index) => index !== action.number)
      return {
        ...state,
        cells: filteredCells,
        tableFooter: getAverageValues(filteredCells),
      }

    case 'INCREMENT':
      const cells = [...state.cells]
      const incrementedCell = cells[action.rowIndex].find((cell) => cell.id === action.id);
      if (incrementedCell) {
        incrementedCell.amount++;
      }
      return {
        ...state,
        ...cells,
        tableFooter: getAverageValues(cells),
      }

      case 'PERCENTS_TOGGLE':
        const percentsCells = [...state.cells]
        percentsCells[action.rowIndex] = percentsCells[action.rowIndex].map((cell) => ({
          ...cell,
          isPercentsShown: !cell.isPercentsShown
        }))
        return {
          ...state,
          cells: percentsCells,
        }

      case 'SET_NUMBER_OF_CLOSEST':
        return {
          ...state,
          numberOfClosest: action.numberOfClosest,
        }

      case 'SHOW_CLOSEST':
        const closest = findClosest(state.cells, action.targetId, state.numberOfClosest);
        const closestCells = [...state.cells];
        closestCells.forEach(row => {
          row.forEach((cell) => {
            if (closest.includes(cell)) {
              cell.isCloser = !cell.isCloser
            }
          })
        })
        return {
          ...state,
          cells: closestCells,
        };

    default:
      return state;
  }
};

export const AppContext = React.createContext(initialState)
