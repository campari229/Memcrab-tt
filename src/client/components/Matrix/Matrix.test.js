import React from 'react';
import { Matrix } from './Matrix';
import TestRender from 'react-test-renderer';

import { createStore } from 'redux';
import { reducer } from '../../store/store';
import { Provider } from 'react-redux';
import { setCells, setNumberOfClosest } from '../../store/store';

const cells = [
  [{ amount: 432, id: 1, isPercentShown: false, isClosest: false }, { amount: 123, id: 2, isPercentShown: false, isClosest: false }, { amount: 534, id: 3, isPercentShown: false, isClosest: false }],
  [{ amount: 228, id: 6, isPercentShown: false, isClosest: false }, { amount: 322, id: 5, isPercentShown: false, isClosest: false }, { amount: 152, id: 4, isPercentShown: false, isClosest: false }],
  [{ amount: 122, id: 7, isPercentShown: false, isClosest: false }, { amount: 899, id: 8, isPercentShown: false, isClosest: false }, { amount: 356, id: 9, isPercentShown: false, isClosest: false }]
]

describe('matrix should work correct', () => {
  let component

  const store = createStore(reducer);
  store.dispatch(setCells(cells))
  store.dispatch(setNumberOfClosest(1));


  TestRender.act(() => {
    component = TestRender.create(
      <Provider store={store}>
        <Matrix />
      </Provider>
    )
  })

it('initial render test', () => {
  expect(component.toJSON()).toMatchSnapshot();
  })

  it('remove button should remove row', () => {
    const button = component.root.findByProps({ id: "remove-btn0" });

    TestRender.act(() => {
      button.props.onClick()
      return undefined
    })

    expect(component.root.findAllByProps({ className: "table__button-remove" }).length).toBe(2)
  });

  it('should increment on click', () => {
    const cell = component.root.findByProps({ id: '0-0' })
    const cellText = component.root.findByProps({ id: 'text:0-0' })

    TestRender.act(() => {
      cell.props.onClick()
      return undefined;
    })

    expect(cellText.props.children).toEqual(229);
  })

  it('should show and hide closest', () => {
    const cell = component.root.findByProps({ id: '1-1' })
    const closestCell = component.root.findByProps({ id: '1-2' })
    const closestText = component.root.findByProps({ id: 'text:1-2' })

    TestRender.act(() => {
      cell.props.onMouseOver()
      return undefined
    })

    expect(closestCell.props.className).toEqual("table__cell table__cell-closest")

    TestRender.act(() => {
      cell.props.onMouseOut()
      return undefined
    })

    expect(closestCell.props.className).toEqual("table__cell")
  })

  it('should show percents', () => {
    const percentsCell = component.root.findByProps({ id: 'percents0' });
    const cell = component.root.findByProps({ id: "text:0-0" });

    TestRender.act(() => {
      percentsCell.props.onMouseOver()
    })

    expect(cell.props.children).toEqual("33%")

    TestRender.act(() => {
      percentsCell.props.onMouseOut()
    })

    expect(cell.props.children).toEqual(229)
  })

  it('add row button should add a row', () => {
      const button = component.root.findByProps({ id: "add-button" })

      TestRender.act(() => {
        button.props.onClick()
      })

      expect(component.root.findAllByProps({ className: "table__row" }).length).toBe(3)
    })
})
