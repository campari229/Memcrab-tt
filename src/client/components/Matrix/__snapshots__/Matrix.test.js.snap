// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matrix should work correct initial render test 1`] = `
<table
  className="table"
>
  <tbody
    className="table__body"
  >
    <tr
      className="table__row"
    >
      <td
        className="table__button-wrapper"
      >
        <button
          className="table__button-remove"
          id="remove-btn0"
          onClick={[Function]}
        >
          Remove row
        </button>
      </td>
      <td
        className="table__cell"
        id="0-0"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:0-0"
        >
          432
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="0-1"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:0-1"
        >
          123
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="0-2"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:0-2"
        >
          534
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="percents0"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        1089
      </td>
    </tr>
    <tr
      className="table__row"
    >
      <td
        className="table__button-wrapper"
      >
        <button
          className="table__button-remove"
          id="remove-btn1"
          onClick={[Function]}
        >
          Remove row
        </button>
      </td>
      <td
        className="table__cell"
        id="1-0"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:1-0"
        >
          228
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="1-1"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:1-1"
        >
          322
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="1-2"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:1-2"
        >
          152
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="percents1"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        702
      </td>
    </tr>
    <tr
      className="table__row"
    >
      <td
        className="table__button-wrapper"
      >
        <button
          className="table__button-remove"
          id="remove-btn2"
          onClick={[Function]}
        >
          Remove row
        </button>
      </td>
      <td
        className="table__cell"
        id="2-0"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:2-0"
        >
          122
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="2-1"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:2-1"
        >
          899
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="2-2"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <p
          className="table__text"
          id="text:2-2"
        >
          356
        </p>
        <div
          className="table__percents"
          style={
            Object {
              "height": "0",
            }
          }
        />
      </td>
      <td
        className="table__cell"
        id="percents2"
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        1377
      </td>
    </tr>
  </tbody>
  <tfoot
    className="table__footer"
  >
    <tr
      className="table__row-footer"
    >
      <td
        className="table__cell-avarange"
      >
        261
      </td>
      <td
        className="table__cell-avarange"
      >
        448
      </td>
      <td
        className="table__cell-avarange"
      >
        347
      </td>
      <td
        className="table__button-wrapper"
      >
        <button
          className="table__button-new"
          id="add-button"
          onClick={[Function]}
        >
          add row
        </button>
      </td>
    </tr>
  </tfoot>
</table>
`;
